Keas.Mvc/ClientApp/components/Equipment/EquipmentAttribute.tsx

```diff
@@ -55,23 +55,19 @@ export default class EquipmentAttribute extends React.Component<IProps, {}> {
         return (
             <Typeahead
                 id={`attribute-${this.props.index}`} // for accessibility
-                allowNew={true}
+                allowNew={false}
                 disabled={this.props.disabledEdit}
                 options={this.props.commonKeys}
                 selected={this.props.attribute.key ? [this.props.attribute.key] : []}
-                onChange={(selected: any) => {
+                onChange={selected => {
                     if (selected && selected.length === 1) {
-                        if (!!selected[0].label) {
-                            // if user selects new from dropdown, it is an object
-                            this.props.changeProperty(this.props.index, "key", selected[0].label);
-                        } else {
-                            // otherwise, it's a string
-                            this.props.changeProperty(this.props.index, "key", selected[0]);
+                        this.props.changeProperty(this.props.index, "key", selected[0]);
+                    } else {
+                        if (!selected || selected && selected.length === 0) {
```

Comment:
@jSylvestre this reads funny and also the second part isn't needed.  `if (!selected || selected.length === 0)` does the same thing, because if `!selected` it true then the second part doesn't get evaluated, so by the same you get to `selected && selected.length === 0` then you already know `selected` is truthy.  unless i'm missing something you caught in practice.